crypto/signature_nocgo.go

```diff
@@ -71,6 +75,33 @@ func Sign(hash []byte, prv *ecdsa.PrivateKey) ([]byte, error) {
 	return sig, nil
 }
 
+// VerifySignature checks that the given public key created signature over hash.
+// The public key should be in compressed (33 bytes) or uncompressed (65 bytes) format.
+// The signature should have the 64 byte [R || S] format.
+func VerifySignature(pubkey, hash, signature []byte) bool {
+	if len(signature) != 64 {
+		return false
+	}
+	sig := &btcec.Signature{R: new(big.Int).SetBytes(signature[:32]), S: new(big.Int).SetBytes(signature[32:])}
+	key, err := btcec.ParsePubKey(pubkey, btcec.S256())
+	if err != nil {
+		return false
```

Comment:
Another design question: we're throwing away error-info here, is that really needed?

Reply:
btcec is the fallback for platforms that can't link C code. It provides better errors because it's Go code. There is no good way to get those errors from the C version. IMHO it doesn't matter much: the signature is invalid if any of the inputs are.

Reply:
Maybe more context: I added this because I want a function that does decompression and verification in one call without any pubkey/signature decoding. We get the compressed pubkey  in ENR and need to verify that it created the signature. IMHO it's best to just pass these values into libsecp256k1.

We could go ahead and make it so `secp256k1_ecdsa_verify_enc` returns different ints for 'invalid signature encoding', 'invalid pubkey encoding' and 'signature mismatch', but it'll just make the code more complicated. Callers won't care: all they want to know is whether the signature checks out.